### PR TITLE
fix(select): aligned select underline, placeholder and text with input

### LINKED ISSUE #2124: Input components (md-input, md-select, md-checkbox etc) aren't aligned in a flex container...
... without setting `align-items: center`

**Updated Plunk: http://plnkr.co/edit/TgFy04UErh9Xhr0A4XzK?p=preview**

`md-input` behaves differently to `md-select` (and at least `md-checkbox`, possibly all the other "input" components) when the elements are within a "flex container"

See this plunk for an example: http://plnkr.co/edit/aLXsp8MlV2AoOkC9aEFP?p=preview

This issue might be resolved with the in-progress refactor of input?

### PR DESCRIPTION
**Pull request by EladBezalel**
- added ellipsis to select placeholder
- aligned floating placeholder with input
- support font-size

fixes #2124

>**Note**: the next PR would support select font-size in panel

### CODE REVIEW

File: src/lib/select/select.html [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-    [style.width.px]="_selectedValueWidth"> {{ placeholder }} </span>
-  <span class="mat-select-value" *ngIf="_selectionModel.hasValue()">
-    <span class="mat-select-value-text">{{ triggerValue }}</span>
-  </span>
+    [style.visibility]="_getPlaceholderVisibility()">
+    <span>{{ placeholder }}</span>
+  </div>
+  <div class="mat-select-value-wrapper" >
+    <span class="mat-select-value" *ngIf="_selectionModel.hasValue()">
+      {{ triggerValue }}
```
Comment (jelbourn) [inline src/lib/select/select.html]: Remove spaces from inside the `{{ }}` while you're in here?

File: src/lib/core/style/_input-common.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
+@import './variables';
+
```
Comment (jelbourn) [inline src/lib/core/style/_input-common.scss]: Add top-level description comment that mentions these are styles used by input-style components, including md-input-container and md-select

**Comment by EladBezalel** [comment]
@jelbourn @kara @mmalerba can you please test and review?

**Comment by mmalerba** [comment]
It looks like this still causes the input to be misaligned when placed with other components:
misaligned-input

It should look like this: 
correct-input-align

Also, it looks like the text for the input and select is correctly aligned, but it still looks strange since they have different spacing between the text and the underline:
misaligned-select

**Comment by EladBezalel** [comment]
@mmalerba where did you tested it? 
The underline was perfect on win10 all browsers

**Comment by mmalerba** [comment]
This was on Ubuntu (both Chrome & Firefox)

**Comment by kara** [comment]
@EladBezalel I'm seeing the same baseline mismatch between inputs and textareas on Mac + Chrome. Can you double-check?

**Comment by willshowell** [comment]
Worth noting that @mmalerba's second comment:
> Also, it looks like the text for the input and select is correctly aligned, but it still looks strange since they have different spacing between the text and the underline:
> 

...is true for me on OSX Safari but not OSX chrome

**Comment by exports** [comment]
Any update on this PR? After two months md-select still does not play well with md-input-container.

**Comment by jelbourn** [comment]
Going to do this by moving `md-select` to live inside `md-form-field` the same as input

**Pull request closed without merging**